src/graph.rs

```diff
@@ -2079,8 +2079,25 @@ pub(crate) fn parse_js_module_from_module_info(
   let mut module = JsModule::new(specifier.clone(), source);
   module.media_type = media_type;
 
-  // Analyze the TypeScript triple-slash references
+  // Analyze the TypeScript triple-slash references and self types specifier
   if graph_kind.include_types() {
+    if is_untyped(&module.media_type) {
```

Comment:
Maybe we should get rid of this condition since `module_info.self_types_specifier` will never be set for unsupported media types?